docs/design/014-did-doc/README.md

```diff
@@ -0,0 +1,590 @@
+```
+
+#### Register did method based on Indy pool with did resolver
+
+```Rust
+// Register did method based on Indy pool with did resolver.
+//
+// did_resolver_handle: resolver handle
+// method: method name, for example, 'sov'
+// pool_handle: pool_handle
+// options: some options
+// returns - ()
+pub fn indy_register_indy_did_method(did_resolver_handle,
+                                     method,
+                                     pool_handle,
+                                     options) -> Future<()>
+```
+
+#### Register custom did method with did resolver
+
+```Rust
+// Register custom did method with did resolver.
+//
+// did_resolver_handle: resolver handle
+// method: method name, for example, 'custom'
+// ..callbacks: list of callbacks that define method behaviour
+// options: some options
+// returns - ()
+pub fn indy_register_did_method(did_resolver_handle,
+                                method,
+                                ..callbacks,
+                                options) -> Future<()>
+```
+
+#### Close did resolver and releases all related resources
+
+```Rust
+// Close did resolver and releases all related resources.
+//
+// did_resolver_handle: resolver handle
+// returns - ()
+pub fn indy_close_did_resolver(did_resolver_handle) -> Future<()>
+```
+
+### Resolving DID Doc
+
+#### Resolve DID Doc for DID
+
+```Rust
+// Resolve DID Doc for DID.
+//
+// did_resolver_handle: resolver handle
+// did: did to resolve
+// options: some options, like cache behaviour
+// returns - did_doc: resolved DID Doc
+pub fn indy_resolve_did_doc(did_resolver_handle,
+                            did,
+                            options) -> Future<did_doc>
+```
+
+### DID and DID keys creation
+
+#### Creates a new DID
+
+```Rust
+// Create a new DID.
+//
+// Note: It doesn't create any records in the wallet.
+//
+// did_resolver_handle: resolver handle
+// method: did method to use
+// options: some options
+// returns - did: Generated DID
+pub fn indy_create_did(did_resolver_handle,
+                       method,
+                       options) -> Future<did>
+```
+
+#### Create a new key pair associated with DID
+
+```Rust
+// Create a new key pair associated with DID.
+//
+// It verifies that key options are compatible with method,
+// creates and stores key in the wallet, creates and stores association
+// of this key with did.
+//
+// Note: It can produce the same data as indy_create_my_did.
+//
+// did_resolver_handle: resolver handle
+// did: did to associate with this key
+// options: some options like key type and seed
+// returns - key_json: json that describes the key. Can be passed to did
+//                     doc builder
+pub fn indy_create_did_key(did_resolver_handle,
+                           did,
+                           options) -> Future<key_json>
+```
+
+### DID Doc creation, updating and publishing
+
+#### Create a new DID Doc builder instance
```

Comment:
Do methods `indy_did_doc_*_public_key`, `indy_did_doc_*_authorization` and `indy_did_doc_*_service_endpoint` make changes to the DID Doc in wallet? Or do they maintain in-memory delta of changes after each such call?

Reply:
We don't store DID Doc in wallet at all. Wallet only stores keys related to DID.